builtin/logical/mysql/secret_creds.go

```diff
@@ -59,27 +67,58 @@ func (b *backend) secretCredsRevoke(
 		return nil, err
 	}
 
+	// Get the role name
+	rolenameRaw, ok := req.Secret.InternalData["rolename"]
+	if !ok {
+		return nil, fmt.Errorf("secret is missing rollname internal data")
+	}
+	rolename, ok := rolenameRaw.(string)
+
+	role, err := b.Role(req.Storage, rolename)
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return logical.ErrorResponse(fmt.Sprintf("unknown role: %s", rolename)), nil
+	}
+
 	// Start a transaction
 	tx, err := db.Begin()
 	if err != nil {
 		return nil, err
 	}
 	defer tx.Rollback()
 
-	// Revoke all permissions for the user. This is done before the
-	// drop, because MySQL explicitly documents that open user connections
-	// will not be closed. By revoking all grants, at least we ensure
-	// that the open connection is useless.
-	_, err = tx.Exec("REVOKE ALL PRIVILEGES, GRANT OPTION FROM '" + username + "'@'%'")
-	if err != nil {
-		return nil, err
+	// Check for an empty revokeSQL string
+	// set it to a default query if the string is empty
+	if role.RevokeSQL == "" {
+		role.RevokeSQL = "REVOKE ALL PRIVILEGES, GRANT OPTION FROM '" + username + "'@'%'; DROP USER '" + username + "'@'%'"
 	}
 
-	// Drop this user. This only affects the next connection, which is
-	// why we do the revoke initially.
-	_, err = tx.Exec("DROP USER '" + username + "'@'%'")
-	if err != nil {
-		return nil, err
+	for _, query := range strutil.ParseArbitraryStringSlice(role.RevokeSQL, ";") {
+		query = strings.TrimSpace(query)
+		if len(query) == 0 {
+			continue
+		}
+
+		// convert {{name}} to username value because we can't use
+		// prepared statements for REVOKE and DROP commands
+		// Generates the following error
+		// 1295: This command is not supported in the prepared statement protocol yet
+		// Reference https://mariadb.com/kb/en/mariadb/prepare-statement/
+		query = strings.Replace(query, "{{name}}", username, -1)
+
+		// Revoke all permissions for the user. This is done before the
```

Comment:
These comments here make no sense because they apply to the default SQL. You should refactor the comments when you pull out the revocation sql into a const and place them as comments on that value.